File: 1_Review/python/3_importing_modules.py

# import entire module
import random
items = ['A','B','C','D','E']
print('***available choices to choose from***')
print(items)
print('***first random choice***')
print(random.choice(items))
print('***second random choice***')
print(random.choice(items))
print()

# import entire module and give it an alias
import random as m
items = ['A','B','C','D','E']
print('***before shuffle***')
print(items)
m.shuffle(items)
print('***after shuffle***')
print(items)
print()

# importing specific object from a module
from math import pow
print(f'2 to the power of 2 is: {pow(2,2)}')

File: 1_Review/javascript/3_importing_modules.js
// 3) Importing modules
// Run: node 3_importing_modules.js

import { randomInt } from 'node:crypto';

const items = ['A', 'B', 'C', 'D', 'E'];

function choice(array) {
  return array[randomInt(0, array.length)];
}

function shuffleInPlace(array) {
  // Fisher-Yates shuffle
  for (let i = array.length - 1; i > 0; i -= 1) {
    const j = randomInt(0, i + 1);
    [array[i], array[j]] = [array[j], array[i]];
  }
}

console.log('***available choices to choose from***');
console.log(items);
console.log('***first random choice***');
console.log(choice(items));
console.log('***second random choice***');
console.log(choice(items));
console.log('');

const items2 = ['A', 'B', 'C', 'D', 'E'];
console.log('***before shuffle***');
console.log(items2);
shuffleInPlace(items2);
console.log('***after shuffle***');
console.log(items2);
console.log('');

// Importing specific things isn't always needed in JS (Math is global)
console.log(`2 to the power of 2 is: ${2 ** 2}`);
